### PR TITLE
Added missing sendgrid readme

### LINKED ISSUE #12217: Sendgrid provider is missing README.md file
The SendGrid provider is missing a README.md file -- this means we haven't yet published a release for this.

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/sendgrid/PROVIDER_CHANGES_0.0.1.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+
+### Release 1.0.0b2
```
Comment (ashb) [inline airflow/providers/sendgrid/PROVIDER_CHANGES_0.0.1.md]: This file and the version number don't match  -- does that matter?

Comment (potiuk) [reply airflow/providers/sendgrid/PROVIDER_CHANGES_0.0.1.md]: Yep. I have to add proper versioning support (that's one of the remaining tasks)

File: airflow/providers/sendgrid/PROVIDER_CHANGES_1.0.0b2.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+
+### Release 1.0.0b2
```
Comment (ashb) [inline airflow/providers/sendgrid/PROVIDER_CHANGES_1.0.0b2.md:3]:
```suggestion
### Release 1.0.0b1
```

In README and filename too please.

**Comment by potiuk** [comment]
Ah sorry. I had  a meeting

**Pull request closed without merging**